app/controllers/urls_controller.rb

```diff
@@ -0,0 +1,85 @@
+class UrlsController < ApplicationController
+  before_action :logged_in_user
+  before_action :correct_user, only: [:edit, :update]
+
+
+  def new
+    @url = Url.new
+  end
+
+  def create
+    @url = Url.new(url_params)
+    @url.user = current_user
+    if @url.save
+      flash[:success] = "Shortened!"
+      render 'land'
+    else
+      flash[:danger] = @url.errors.full_messages.first
+      render 'new'
+    end
+  end
+
+  def land
+    @url = Url.find(params[:id])
+  end
+
+  def index
+    @urls = Url.paginate(page: params[:page])
+    @pageTitle = "List of Links"
+  end
+
+  def myIndex
```

Comment:
Check out [nested resources](http://guides.rubyonrails.org/routing.html#nested-resources). This could be nested under a given user like `/users/123/urls`.

But you only want to expose the current user's urls, and you know who they are by the session. So, it could be `/current_user/urls` or  `/my/urls`.